### PR TITLE
Fix <tplName>Config.cmake files to not point into build dir (#500)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This PR contains the updated test exposing the defect reported in #500 and the simple code changes to fix them.

### CODE REVIEW

**Review by rabartlett1972 — APPROVED** [review]
Because of the urgency of this defect, I will go ahead and allow this to merge to TriBITS 'master' so that I can create a Trilinos PR with an updated TriBITS snapshot ASAP.

**Review by KyleFromKitware — COMMENTED** [review]
LGTM

File: test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake [hunk 1]
```diff
@@ -159,7 +168,13 @@ function(TribitsExampleProject2_find_tpl_parts  sharedOrStatic  findingTplsMetho
     ${cmakePrefixPathEnvArg}
 
     TEST_0
+      MESSAGE "Copy TribitsExampleProject2 so we can delete it after the install"
+      CMND cp
```
[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake:172]: I think there was some strange error with `cmake -E copy` with an older version of CMake in the past and this `cp` command just got copied and pasted around.  To make these tests run on Windows, we will have to address issues like this.  (The code in  TriBITS is just raw CMake for the basic configure, build, and test but many of the tests can only run on Unix envs.)

[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake:172]: When I get to setting up a GitHub Actions build on Windows, I will be looking into these issues naturally then to see how much of the TriBITS test suite can be made to run seamlessly on Windows as well.  (There are some very intrusive tests that use tools like objdump for checking RPATH or grep specific output from build tools like  GNU Make or Ninja that will be impossible or not be worth porting.)

**Pull request merged**